Library/Homebrew/cask/lib/hbc/quarantine.rb

```diff
@@ -0,0 +1,70 @@
+require "development_tools"
+module Hbc
+  module Quarantine
+    module_function
+
+    QUARANTINE_ATTRIBUTE = "com.apple.quarantine".freeze
+
+    QUARANTINE_SCRIPT = "#{HOMEBREW_LIBRARY_PATH}/cask/lib/hbc/utils/quarantine.swift".freeze
+
+    # @private
+    def swift
+      @swift ||= DevelopmentTools.locate("swift")
+    end
+
+    def available?
+      status = !swift.nil?
+      odebug "Quarantine is #{status ? "available" : "not available"}."
+      !swift.nil?
+    end
+
+    def detect(file)
+      odebug "Verifying Gatekeeper status of #{file}"
+
+      quarantine_status = !status(file).empty?
+
+      odebug "#{file} is #{quarantine_status ? "quarantined" : "not quarantined"}"
+
+      quarantine_status
+    end
+
+    def status(file, command = SystemCommand)
+      command.run("/usr/bin/xattr",
+                  args:        ["-p", QUARANTINE_ATTRIBUTE, file],
+                  print_stderr: false).stdout
+    end
+
+    def cask(cask, downloaded_path, command = SystemCommand)
+      odebug "Quarantining #{downloaded_path}"
+      quarantiner = command.run(swift,
+                                args: [
+                                  QUARANTINE_SCRIPT,
+                                  downloaded_path,
+                                  cask.url.to_s,
+                                  cask.homepage.to_s,
+                                ])
+
+      return if quarantiner.success?
+
+      raise CaskError, "when quarantining #{downloaded_path}: #{quarantiner.stderr}"
```

Comment:
`"Failed to quarantine #{downloaded_path}: #{quarantiner.stderr}"`

Or maybe create a `QuarantineError` class, since this format is already used multiple times.

Reply:
> Or maybe create a QuarantineError class, since this format is already used multiple times.

Done :+1: